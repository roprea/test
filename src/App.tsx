import { BrowserRouter } from 'react-router-dom';
// import { InteractionType, PublicClientApplication } from '@azure/msal-browser';
// import type { Configuration } from '@azure/msal-browser';
// import { MsalAuthenticationTemplate, MsalProvider } from '@azure/msal-react';
import Content from 'components/Content';
import Header from 'components/Header';

// const configuration: Configuration = {
//   auth: {
//     clientId: String(process.env.REACT_APP_CLIENT_ID),
//     authority: 'https://login.microsoftonline.com',
//     redirectUri: window.location.origin,
//   },
// };
// const application = new PublicClientApplication(configuration);

function App() {
  return (
    // <MsalProvider instance={application}>
      <BrowserRouter>
        {/* <MsalAuthenticationTemplate interactionType={InteractionType.Redirect}> */}
          <Header />
          <Content />
        {/* </MsalAuthenticationTemplate> */}
      </BrowserRouter>
    // </MsalProvider>
  );
}

export default App;
