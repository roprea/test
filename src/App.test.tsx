import type { ReactNode } from 'react';
import { basicTestRunner } from 'test-utils/basic-test-runner';
import { render, screen } from 'test-utils/testing-library';
import App from './App';

jest.mock('@azure/msal-react', () => {
  const actual = jest.requireActual('@azure/msal-react');

  return {
    ...actual,
    MsalAuthenticationTemplate: ({ children }: { children: ReactNode }) => <>{children}</>,
    MsalProvider: ({ children }: { children: ReactNode }) => <>{children}</>,
  };
});
jest.mock('components/Header', () => ({
  __esModule: true,
  default: () => <div>Application header</div>,
}));
jest.mock('components/Content', () => ({
  __esModule: true,
  default: () => <div>Application content</div>,
}));

describe('App', () => {
  basicTestRunner(<App />, () => ({ only: ['snapshot', 'accessibility'] }));

  it('render application header', () => {
    render(<App />);
    expect(screen.getByText('Application header')).toBeInTheDocument();
  });

  it('render application content', () => {
    render(<App />);
    expect(screen.getByText('Application content')).toBeInTheDocument();
  });
});
