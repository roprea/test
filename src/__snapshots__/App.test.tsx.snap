// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Basic component behaviour should match snapshot 1`] = `
<div>
  <div>
    Application header
  </div>
  <div>
    Application content
  </div>
</div>
`;
